### PR TITLE
Copy in interfaces rather than using encoding module import

### PR DESCRIPTION
**Pull request by bz2**
Workaround odd gccgo issue, which fails with errors along the lines of:

../xml/marshal.go:10:2: cannot find package "encoding" in any of:
	/usr/src/pkg/encoding (from $GOROOT)

As this forked code only imports encoding for the TextMarshaler and TextUnmarshaler interface, just bring those in where they're used and drop the import.

### CODE REVIEW

File: marshal.go [hunk 1]
```diff
@@ -123,6 +122,14 @@ func MarshalIndent(v interface{}, prefix, indent string) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
+// TextMarshaler is the interface implemented by an object that can
+// marshal itself into a textual form.
+//
+// MarshalText encodes the receiver into UTF-8-encoded text and returns the result.
+type TextMarshaler interface {
```
Comment (rogpeppe) [inline marshal.go]: let's not export this.

File: read.go [hunk 2]
```diff
@@ -14,6 +13,16 @@ import (
 	"strings"
 )
 
+// TextUnmarshaler is the interface implemented by an object that can
+// unmarshal a textual representation of itself.
+//
+// UnmarshalText must be able to decode the form generated by MarshalText.
+// UnmarshalText must copy the text if it wishes to retain the text
+// after returning.
+type TextUnmarshaler interface {
```
Comment (rogpeppe) [inline read.go]: ditto

**Comment by rogpeppe** [comment]
LGTM with two trivial changes, thanks!

**Comment by bz2** [comment]
Have done the lowercasing as suggested.

**Pull request merged**